online-banking-cz/src/main/java/com/adyen/checkout/onlinebankingcz/OnlineBankingCZDelegate.kt

```diff
@@ -0,0 +1,71 @@
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by atef on 25/8/2022.
+ */
+
+package com.adyen.checkout.onlinebankingcz
+
+import com.adyen.checkout.components.PaymentComponentState
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.payments.request.OnlineBankingCZPaymentMethod
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.issuerlist.IssuerListDelegate
+import com.adyen.checkout.issuerlist.IssuerListInputData
+import com.adyen.checkout.issuerlist.IssuerListOutputData
+import com.adyen.checkout.issuerlist.IssuerModel
+import com.adyen.checkout.issuerlist.IssuersUtils
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class OnlineBankingCZDelegate(
+    private val paymentMethod: PaymentMethod,
+    private val onlineBankingCZPaymentMethod: () -> OnlineBankingCZPaymentMethod
+) : IssuerListDelegate<OnlineBankingCZPaymentMethod> {
+
+    private val _outputDataFlow = MutableStateFlow<OnlineBankingCZOutputData?>(null)
+    override val outputDataFlow: Flow<OnlineBankingCZOutputData?> get() = _outputDataFlow
+
+    private val _componentStateFlow = MutableStateFlow<PaymentComponentState<OnlineBankingCZPaymentMethod>?>(null)
+    override val componentStateFlow: Flow<PaymentComponentState<OnlineBankingCZPaymentMethod>?>
+        get() = _componentStateFlow
+
+    override fun getIssuers(): List<IssuerModel> {
+        return paymentMethod.issuers?.let {
+            IssuersUtils.getIssuers(it)
+        } ?: IssuersUtils.getLegacyIssuers(paymentMethod.details)
+    }
+
+    override fun createComponentState(outputData: IssuerListOutputData) {
+        val issuerListPaymentMethod = onlineBankingCZPaymentMethod()
+        issuerListPaymentMethod.type = getPaymentMethodType()
+        issuerListPaymentMethod.issuer = outputData.selectedIssuer?.id ?: ""
+
+        val paymentComponentData = PaymentComponentData<OnlineBankingCZPaymentMethod>()
+        paymentComponentData.paymentMethod = issuerListPaymentMethod
+
+        val state = PaymentComponentState(paymentComponentData, outputData.isValid, true)
+        _componentStateFlow.tryEmit(state)
+    }
+
+    override fun onInputDataChanged(inputData: IssuerListInputData) {
+        val outputData = OnlineBankingCZOutputData(TERMS_CONDITIONS_URL, inputData.selectedIssuer)
+
+        _outputDataFlow.tryEmit(outputData)
+
+        createComponentState(outputData)
+    }
+
+    override fun getPaymentMethodType(): String {
+        return paymentMethod.type ?: PaymentMethodTypes.UNKNOWN
+    }
+
+    fun getTermsAndConditionsUrl(): String = TERMS_CONDITIONS_URL
+
+    companion object {
+        private const val TERMS_CONDITIONS_URL = "https://static.payu.com/sites/terms/files/payu_privacy_policy_cs.pdf"
```

Comment:
Why is the URL hardcoded? I don't think it's a good idea. What if this URL ever changes? In that case we have to change the SDK, publish a new release, merchants have to update and release a new version of their app. If users can't view terms and conditions merchants might be in trouble, because of laws. 🤔  It would be better to have this coming from the backend

Reply:
good point and I asked same questions before and had the same thoughts about better to come from backend 😣 and they just told me the link will stay the same maybe the file will be updated on same link